ui/src/dropdown.test.tsx

```diff
@@ -375,5 +375,40 @@ describe('Dropdown.tsx', () => {
       fireEvent.click(getByTestId(name))
       expect(getAllByRole('listitem')).toHaveLength(40) // Fluent Detaillist uses virtualization, so only first 40 listitems are rendered.
     })
+
+    it(`Displays dialog when choices > 100 and 'popup' prop is not provided`, () => {
+      const { getByTestId, getByRole  } = render(<XDropdown model={dialogProps} />)
+
+      fireEvent.click(getByTestId(name))
+
+      expect(getByRole('dialog')).toBeInTheDocument()
+    })
+
+    it(`Displays dialog when choices > 100 and 'popup' prop is set as 'auto'`, () => {
+      const { getByTestId, getByRole  } = render(<XDropdown model={dialogProps} />)
+
+      fireEvent.click(getByTestId(name))
+
+      expect(getByRole('dialog')).toBeInTheDocument()
+    })
+
+    it(`Displays dialog when choices < 100 and 'popup' prop is set as 'always'`, () => {
+      dialogProps.popup = 'always'
+      dialogProps.choices = [{ name: 'A' }]
+      const { getByTestId, getByRole  } = render(<XDropdown model={dialogProps} />)
+
+      fireEvent.click(getByTestId(name))
+
+      expect(getByRole('dialog')).toBeInTheDocument()
+    })
+
+    it(`Displays dialog when choices > 100 and 'popup' prop is set as 'never'`, () => {
```

Comment:
If `never` is specified, the dialog shouldn't be displayed.

Reply:
Fixed.